docs/concepts/gardenlet.md

```diff
@@ -309,20 +309,32 @@ If there are no extension resources anymore, its status will be `False`.
 
 This condition is taken into account by the `ControllerRegistration` controller part of `gardener-controller-manager` when it computes which extensions have to be deployed to which seed cluster. See [Gardener Controller Manager](controller-manager.md#controllerregistration-controller) for more details.
 
-### [`VPAEvictionRequirements` Controller](../../pkg/gardenlet/controller/vpaevictionrequirements)
+### [`Gardenlet` Controller](../../pkg/gardenlet/controller/gardenlet)
 
-The `VPAEvictionRequirements` controller in the `gardenlet` reconciles `VerticalPodAutoscaler` objects labeled with `autoscaling.gardener.cloud/eviction-requirements: managed-by-controller`. It manages the [`EvictionRequirements`](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler/enhancements/4831-control-eviction-behavior) on a VPA object, which are used to restrict when and how a Pod can be evicted to apply a new resource recommendation.
-Specifically, the following actions will be taken for the respective label and annotation configuration:
-* If the VPA has the label `eviction-requirements.autoscaling.gardener.cloud/downscale-restriction: never`, an `EvictionRequirement` is added to the VPA object that allows evictions for upscaling only
-* If the VPA has the label `eviction-requirements.autoscaling.gardener.cloud/downscale-restriction: in-maintenance-window-only`, the same `EvictionRequirement` is added to the VPA object when the Shoot is currently outside of its maintenance window. When the Shoot is inside its maintenance window, the `EvictionRequirement` is removed. Information about the Shoot maintenance window times are stored in the annotation `shoot.gardener.cloud/maintenance-window` on the VPA
+The `Gardenlet` controller in the `gardenlet` reconciles a `Gardenlet` resources with the same name as the `Seed` the gardenlet is responsible for.
+This is used to implement self-upgrades of `gardenlet` based on information pulled from the garden cluster.
+For a general overview, see [this document](../deployment/deploy_gardenlet.md).
+
+On `Gardenlet` reconciliation, the controller deploys the `gardenlet` within its own cluster which after downloading the Helm chart specified in `.spec.deployment.helm.ociRepository` and rendering it with the provided values/configuration.
```

Comment:
Self-upgrades are very convenient feature to reduce the maintenance effort. However, the implementation also provides a potential "backdoor" allowing deployment of basically anything into the cluster if I have permission to change the OCI repository in the `Gardenlet` resource.
Do we want to add some means in addition to the existing RBAC to prevent random use?

Reply:
See https://github.com/gardener/gardener/pull/9874#discussion_r1622739265. Only Gardener operators have privileges to manipulate the `Gardenlet` resources. I don't think it makes sense to invest into additional(, rather artifical (for now)) validation. If we go down this path, we have a lot of work to be done, just think about image vector overwrites in the component configurations.